src/layer.js

```diff
@@ -145,50 +145,63 @@ export default class Layer {
    * @return {Promise} [will resolve once image src has been retrieved and dimensions calculated]
    */
   calculateSize() {
-    // Create image element
-    this.imageEl = new Image();
-    this.imageEl.crossOrigin = 'Anonymous';
-    this.imageEl.src = this.data.image;
+    var _this = this;
+    var img = new Image();
+    img.crossOrigin = 'Anonymous';
+    return (new Promise((resolve, reject) => {
+      img.onerror = (err) => reject(err);
+      img.onload = () => resolve(img);
+      img.src = _this.data.image;
+      if (img.complete || img.complete === undefined) {
```

Comment:
`img.complete === undefined` is invalid as undefined is a type, instead use `typeof img.complete === 'undefined'` although it looks like this condition is not required when using `if (img.complete)`